base/sparse/sparsevector.jl

```diff
@@ -1681,3 +1681,35 @@ droptol!(x::SparseVector, tol, trim::Bool = true) = fkeep!(x, (i, x, tol) -> abs
 
 dropzeros!(x::SparseVector, trim::Bool = true) = fkeep!(x, (i, x, other) -> x != 0, nothing, trim)
 dropzeros(x::SparseVector, trim::Bool = true) = dropzeros!(copy(x), trim)
+
+function _fillnonzero!{Tv,Ti}(arr::SparseMatrixCSC{Tv, Ti}, val)
+    m, n = size(arr)
+    arr.colptr = convert(Vector{Ti}, collect(1:m:n*m+1))
+    arr.rowval = convert(Vector{Ti}, vcat([1:m for _ in 1:n]...))
```

Comment:
the terminology is a bit confusing. sparsematrix and sparsevector are composite types, where you very rarely need or want to wholesale replace the arrays they contain. the fields themselves are usually mutable arrays that can be modified, but only as long as it remains a reference to the same object. though being an immutable type would make it tough to modify the dimension parameters which I have needed to do once in a while.